servicetalk-opentelemetry-http/src/test/java/io/servicetalk/opentelemetry/http/ServicetalkSpanStatusExtractorTest.java

```diff
@@ -43,39 +45,64 @@ class ServicetalkSpanStatusExtractorTest {
     @Mock
     HttpResponseMetaData responseMetaData;
 
-    @Test
-    void testStatus200To399() {
-        int executions = 0;
+    @ParameterizedTest(name = "{displayName} [{index}]: isServer={0}")
+    @ValueSource(booleans = {true, false})
+    void testStatus200To399(boolean isServer) {
         for (int code = 100; code < 400; code++) {
+            when(responseMetaData.status()).thenReturn(HttpResponseStatus.of(code, "any"));
+            getExtractor(isServer).extract(spanStatusBuilder, requestMetaData, responseMetaData, null);
+        }
+        // Should remain at the default value of UNSET
+        verify(spanStatusBuilder, times(0)).setStatus(any());
+    }
+
+    @ParameterizedTest(name = "{displayName} [{index}]: isServer={0}")
+    @ValueSource(booleans = {true, false})
+    void testStatus400to499(boolean isServer) {
+        int executions = 0;
+        for (int code = 400; code < 500; code++) {
             executions++;
             when(responseMetaData.status()).thenReturn(HttpResponseStatus.of(code, "any"));
-            ServicetalkSpanStatusExtractor.INSTANCE.extract(spanStatusBuilder, requestMetaData, responseMetaData, null);
+            getExtractor(isServer).extract(spanStatusBuilder, requestMetaData, responseMetaData, null);
+        }
+        if (isServer) {
+            // Should remain at the default value of UNSET
+            verify(spanStatusBuilder, times(0)).setStatus(any());
```

Comment:
That's very interesting behavior for the server-side 😕 
Do you have a link to that convention? Do they provide reasoning why server doesn't need a status in this case?

I can see that they are trying to say that client was at fault for 4xx, so only client-span should have a status. But it still looks weird that some info is not set on the server-side even if we have it. There is also a variation of 2xx status codes that might be useful to track.

Is it a problem if we still set any status?

Reply:
Nvm, I misunderstood what status is meant here. This PR is about `io.opentelemetry.api.trace.StatusCode`, not HTTP status code.

Reply:
I'll clarify that in the PR description since the term `status` is overloaded.